### PR TITLE
week2_project files

### PR DESCRIPTION
**Pull request by srujanreddyj**

### CODE REVIEW

**Comment by srujanreddyj** [comment]
Project Self Assessment

- Do you understand the steps involved in creating and deploying an LTR model? Name them and describe what each step does in your own words.
  - Index the products and queries
  -  Create a LTR store to save models and its versions. 
  - Process a dataset (indexed products and queries) and assign a judgement label either explicitly or implicitly (through click data)
  - Determine features through iterations by logging their feature score and build a featureset. 
  -  Downsample data to produce right mix of training and testing sets
  - Perform training with XGBoost on training set and test against the test set and compare against the base LTR model
  - Continue to perform feature engineering and iterate on featureset and XGBoost parameters with more experimentation.
  - Deploy the model with satisfied scores. 
- What is a feature and featureset?
  - A feature is an attribute that is a measurable piece of data or characterizes the target and act as input to the system.  
  - Feature set is a set of all attributes (features) that we are interested for a problem
- What is the difference between precision and recall?
  - Precision refers to the percentage of positive identifications that were actually correct [fraction of retrieved results that were relevant]
  - Recall refers to the percentage of actual positives that were identified correctly [fraction of relevant results that were retrieved].
- What are some of the traps associated with using click data in your model?
  -  Users click on top n results and low rank results will go with 0 clicks
  - Users can click just for viewing -- out of curiosity.
  - Having bot and other malicious clicks 
  - Users can skip clicking on the results and just scroll leading to not correct click data
- What are some of the ways we are faking our data and how would you prevent that in your application?
  -  We are faking data by faking impressions.  In real world, we would be having the log impression data.
- What is target leakage and why is it a bad thing?
  -  Target leakage or data leakage is when we train our algorithm with data that would not be available at the time of prediction. This leads to bad prediction and having less accuracy metrics.  
- When can using prior history cause problems in search and LTR?
  -  When new items are added/introduced, a cold start and also when there is a complete change in consumers behavior
- Submit your project along with your best MRR scores
   ```
   Simple MRR is 0.294
   LTR Simple MRR is 0.305 
   Hand tuned MRR is 0.648
   LTR Hand Tuned MRR is 0.648
   ```

File: week2/utilities/xgb_utils.py [hunk 1]
```diff
@@ -25,12 +25,24 @@ def plots(xgb_model, xgb_model_name, xgb_feat_map, xgb_plot):
 
 
 # xgb_train_data is a string path to our training file
-def train(xgb_train_data, num_rounds=5, xgb_conf=None ):
-    xgb_params = {'objective': 'reg:logistic'}
-    bst = None
+def train(xgb_train_data, num_rounds=50, xgb_conf=None ):
+    params = {
+        'max_depth': 50,
+        'verbosity': 2,
```
Comment (esvazas) [inline week2/utilities/xgb_utils.py:31]: Cool that you actually experimented with various parameters!

**Review by esvazas — APPROVED** [review]
Looks good! Well done!
